aqandu/api_routes.py

```diff
@@ -337,38 +571,200 @@ def request_model_data_local(lat, lon, radius, start_date, end_date):
     return model_data
 
 
-# Gets data within radius of the provided lat lon within the time frame. The radius units are latlon degrees so this is an approximate bounding circle
-@app.route("/api/request_model_data/", methods=['GET'])
+# returns sensor data for a range of times within a distance radius (meters) of the lat-lon location.  
+@app.route("/api/request_model_data", methods=['GET'])
 def request_model_data():
     query_parameters = request.args
-    lat = query_parameters.get('lat')
-    lon = query_parameters.get('lon')
-    radius = query_parameters.get('radius')
+    try:
+        lat = float(query_parameters.get('lat'))
+        lon = float(query_parameters.get('lon'))
+        radius = float(query_parameters.get('radius'))
+    except ValueError:
+        return 'lat, lon, radius, must be floats.', 400
+
+    start_date = query_parameters.get('start_date')
+    end_date = query_parameters.get('end_date')
+    print("model requuest api with " + str(lat) + ":" + str(lon) + " and radius " + str(radius) + " and start " + str(start_date) + " and end " + str(end_date))
+    # must format these for database
+    start_datetime = utils.parseDateString(start_date)
+    end_datetime = utils.parseDateString(end_date)
+    model_data = request_model_data_local(lat, lon, radius, start_datetime, end_datetime)
+    return jsonify(model_data)
+
+# get estimates within a time frame for a single location
+@app.route("/api/getEstimatesForLocation", methods=['GET'])
+def getEstimatesForLocation():
+    # Check that the arguments we want exist
+    # if not validateInputs(['lat', 'lon', 'estimatesperhour', 'start_date', 'end_date'], request.args):
+    #     return 'Query string is missing one or more of lat, lon, estimatesperhour, start_date, end_date', 400
```

Comment:
I don't think I removed/commented that.  However, it appears whoever did comment it might have found it redundant with the checks for the particular parameters that come below.  I don't have strong feelings one way or the other - except that I think the checks below take care of this.

Reply:
the git blame blames Jack on that one: https://github.com/aqandu/aqandu_live_site/commit/37797f3414c2fa29c00bb2b074be89d419296bbf#diff-a383e0a3c7811331de265700fa93afc2fdfa8059f034dabb1d5046166fe9e17dR357

Reply:
That's funny.  Generally, when Jack complains about things he doesn't like, I blame Oleks!